helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -413,28 +434,58 @@ private void globalRebalance(ResourceControllerDataProvider clusterData,
             HelixRebalanceException.Type.INVALID_CLUSTER_STATUS, ex);
       }
 
-      calculateAndUpdateBaseline(clusterModel, algorithm);
+      final boolean waitForGlobalRebalance = !_asyncGlobalRebalanceEnabled;
```

Comment:
Nit: we flip the boolean a lot here - do you think that's necessary? Can we just use one boolean?

Reply:
It has to be done once, since the inputs for 2 methods are different.
After comparing multiple possible combinations, I feel the current way requires fewer comments.